driver-core/src/test/functional/com/mongodb/client/model/expressions/ExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,646 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import com.mongodb.client.model.Field;
+import com.mongodb.client.model.OperationTest;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonReader;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.Document;
+import org.bson.codecs.BsonDocumentCodec;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.conversions.Bson;
+import org.bson.json.JsonReader;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static com.mongodb.client.model.Aggregates.addFields;
+import static com.mongodb.client.model.expressions.Expressions.CURRENT;
+import static com.mongodb.client.model.expressions.Expressions.ofNull;
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+/**
+ * This test file groups expressions of each type under one test method. Each of
+ * these methods begins by showing how to express literals of the relevant type.
+ * The ensuing assertions then show how to express the computation in Java (when
+ * reasonable), then using the API under test, and then in MQL.
+ */
+@SuppressWarnings({
+        // for demonstration:
+        "PointlessBooleanExpression",
+        "PointlessArithmeticExpression",
+        "ConstantConditions",
+        // for clarity:
+        "Convert2MethodRef"})
+public class ExpressionsFunctionalTest extends OperationTest {
+
+    @Test
+    public void booleanExpressionsTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#boolean-expression-operators
+        // (Complete as of 6.0)
+
+        // literals
```

Comment:
Consider pros and cons of BooleanExpression.of(boolean value)

Reply:
I think `Expressions.of` would be good.

Reply:
We did this.